casanovo/denovo/transformers.py

```
@@ -147,8 +167,17 @@ def __init__(
         n_layers: int = 1,
         dropout: float = 0,
         peak_encoder: PeakEncoder | Callable | bool = True,
+        encoding_type: str = "sinusoidal",
     ):
         """Initialize a SpectrumEncoder."""
+        _VALID_ENCODING_TYPES = {"sinusoidal", "fourier"}
+        if encoding_type not in _VALID_ENCODING_TYPES:
+            raise ValueError(
+                f"Invalid encoding_type '{encoding_type}'. "
+                f"Must be one of: {_VALID_ENCODING_TYPES}"
+            )
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)